ICSharpCode.Decompiler/IL/Transforms/EarlyExpressionTransforms.cs

```diff
@@ -111,5 +111,49 @@ internal static ILInstruction HandleCall(Call inst, ILTransformContext context)
 			}
 			return null;
 		}
+
+		protected internal override void VisitBinaryNumericInstruction(BinaryNumericInstruction inst)
+		{
+			if (inst.Operator != BinaryNumericOperator.BitAnd)
+				return;
```

Comment:
The early return here prevents recursing into the child instructions.
In expression transforms, it's usually better to first call `base.VisitBinaryNumericInstruction()` to handle the children (thus applying transforms in post-order).